pkg/agent/flannel/flannel.go

```diff
@@ -84,8 +85,9 @@ func flannel(ctx context.Context, flannelIface *net.Interface, flannelConf, kube
 	return nil
 }
 
-func LookupExtInterface(iface *net.Interface) (*backend.ExternalInterface, error) {
+func LookupExtInterface(iface *net.Interface, dualStack bool) (*backend.ExternalInterface, error) {
```

Comment:
This might sound like a stylistic gripe but hear me out: what it we passed an enumerated type such as `NetMode`, `IPMode`, or `IfaceMode`? It could have distinct values of `IPv4`, `IPv6`, and `DualStack`. We could chose to make `DualStack` == `IPv4` | `IPv6` or we could chose not to. Regardless the custom type should have helpers functions for asserting if a particular "mode" is set and converting ordinal values to externalizable strings. Maybe this implies more up front work than it is worth but in the longer term I believe that it will maintain readability of the conditional clauses, as proposed, and maximize our ability to understand at-a-glance what mode we are passing at the call site.

Reply:
I think it is a great idea, I agree with you that it would improve things. And I don't think it'll be that hard to implement, let me do it and thanks for the suggestion

Reply:
Still testing, but please have a look. ipv4 and ipv6 have an iota const value